ClientApp/app/(tabs)/create/index.tsx

```diff
@@ -1,25 +1,534 @@
-import axiosInstance from '@/services/axiosInstance';
-import React from 'react';
-import { View, Text, TextInput, Button } from 'react-native';
-const handleSubmit = async () => {{
-  const response = await axiosInstance.get("user-service/user/673114e06f52646e079d1486");
-  console.log('============================================');
-  console.log(response);
-}
-
-}
+import React, { useEffect, useState } from 'react';
+import {
+  View,
+  Text,
+  TextInput,
+  TouchableOpacity,
+  ScrollView,
+  StyleSheet,
+  Modal,
+  FlatList,
+  StatusBar,
+  Alert,
+} from 'react-native';
+import { useForm, Controller, useWatch } from 'react-hook-form';
+import DateTimePicker from '@react-native-community/datetimepicker';
+import { useRouter } from 'expo-router';
+import ConfirmButton from '@/components/ConfirmButton';
+import { IconPlacement } from '@/utils/constants/enums';
+import themeColors from '@/utils/constants/colors';
+import { hs, vs, mhs } from '@/utils/helpers/uiScaler';
+import { API_ENDPOINTS } from '@/utils/api/endpoints';
+import { createEvent } from '@/services/eventService';
+import { useSelector } from 'react-redux';
+
+const sports = ['Soccer', 'Basketball', 'Tennis', 'Swimming', 'Running', 'Football', 'Rugby'];
+
 const Create = () => {
+  const { control, handleSubmit, formState: { errors }, setValue, reset } = useForm<EventFormData>({
+    defaultValues: {
+      eventType: 'public',
+    },
+  });
+  const router = useRouter();
+  const [isSportTypeModalVisible, setSportTypeModalVisible] = useState(false);
+  const [eventDate, setEventDate] = useState(new Date());
+  const [showDatePicker, setShowDatePicker] = useState(false);
+  const [isSuccessModalVisible, setSuccessModalVisible] = useState(false);
+
+  const user = useSelector((state: { user: any }) => state.user);
+
+  useEffect(() => {
+    if (!user) {
+      console.log('User not found');
```

Comment:
I think this useEffect should be taken out, we don't need to log this.

Reply:
agreed! done